src/harness/virtualFileSystemWithWatch.ts

```diff
@@ -777,131 +508,6 @@ interface Array<T> { length: number; [n: number]: T; }`
-                modifiedTime: this.now()
-            };
-        }
-
-        private toFsFile(file: File): FsFile {
-            const fsFile = this.toFsEntry(file.path) as FsFile;
-            fsFile.content = file.content;
-            fsFile.fileSize = file.fileSize;
-            return fsFile;
-        }
-
-        private toFsSymLink(symLink: SymLink): FsSymLink {
-            const fsSymLink = this.toFsEntry(symLink.path) as FsSymLink;
-            fsSymLink.symLink = getNormalizedAbsolutePath(symLink.symLink, getDirectoryPath(fsSymLink.fullPath));
-            return fsSymLink;
-        }
-
-        private toFsFolder(path: string): FsFolder {
-            const fsFolder = this.toFsEntry(path) as FsFolder;
-            fsFolder.entries = [] as FSEntry[] as SortedArray<FSEntry>; // https://github.com/Microsoft/TypeScript/issues/19873
-            return fsFolder;
-        }
-
-        private getRealFsEntry<T extends FSEntry>(isFsEntry: (fsEntry: FSEntry) => fsEntry is T, path: Path, fsEntry = this.fs.get(path)!): T | undefined {
-            if (isFsEntry(fsEntry)) {
-                return fsEntry;
-            }
-
-            if (isFsSymLink(fsEntry)) {
-                return this.getRealFsEntry(isFsEntry, this.toPath(fsEntry.symLink));
-            }
-
-            if (fsEntry) {
-                // This fs entry is something else
-                return undefined;
-            }
-
-            const realpath = this.toPath(this.realpath(path));
-            if (path !== realpath) {
-                return this.getRealFsEntry(isFsEntry, realpath);
-            }
-
-            return undefined;
-        }
-
-        private isFsFile(fsEntry: FSEntry) {
-            return !!this.getRealFile(fsEntry.path, fsEntry);
-        }
-
-        private getRealFile(path: Path, fsEntry?: FSEntry): FsFile | undefined {
-            return this.getRealFsEntry(isFsFile, path, fsEntry);
-        }
-
-        private isFsFolder(fsEntry: FSEntry) {
-            return !!this.getRealFolder(fsEntry.path, fsEntry);
-        }
-
-        private getRealFolder(path: Path, fsEntry = this.fs.get(path)): FsFolder | undefined {
-            return this.getRealFsEntry(isFsFolder, path, fsEntry);
-        }
-
-        fileExists(s: string) {
-            const path = this.toFullPath(s);
-            return !!this.getRealFile(path);
-        }
-
-        getModifiedTime(s: string) {
-            const path = this.toFullPath(s);
-            const fsEntry = this.fs.get(path);
-            return (fsEntry && fsEntry.modifiedTime)!; // TODO: GH#18217
-        }
-
-        setModifiedTime(s: string, date: Date) {
-            const path = this.toFullPath(s);
-            const fsEntry = this.fs.get(path);
-            if (fsEntry) {
-                fsEntry.modifiedTime = date;
-                this.invokeFileAndFsWatches(fsEntry.fullPath, FileWatcherEventKind.Changed);
-            }
-        }
-
         readFile(s: string): string | undefined {

```

Comment:
Shouldnt `readFile` be part of main VFS?

Reply:
Also getFileSize needs to be on main VFS

Reply:
Thanks for catching that. I think I made a mistake when merging from main.